File: BackEnd/PHP/laravelshop/tests/E2E/ResilienceAutoTuneE2ETest.php
<?php

namespace Tests\E2E;

use Illuminate\Foundation\Testing\RefreshDatabase;
use Tests\Support\ApiAuthTestHelpers;
use Tests\TestCase;

class ResilienceAutoTuneE2ETest extends TestCase
{
    use ApiAuthTestHelpers;
    use RefreshDatabase;

    public function test_resilience_policies_endpoint_exposes_tuned_policy_snapshot(): void
    {
        $admin = $this->createApiUser('ADMIN');

        $this->actingAsApiUser($admin)
            ->getJson('/api/v1/resilience/circuit-breakers/policies')
            ->assertOk()
            ->assertJsonPath('data.items.0.name', 'crawler')
            ->assertJsonPath('data.items.0.options.threshold', 5)
            ->assertJsonPath('data.items.0.stats.opened', 1);
    }
}

File: BackEnd/Java/java-spring-maven-jpa-postgresql/src/test/java/com/pbshop/java/spring/maven/jpa/postgresql/e2e/ResilienceAutoTuneE2ETest.java
package com.pbshop.java.spring.maven.jpa.postgresql.e2e;

import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;

import org.junit.jupiter.api.Test;

import com.pbshop.java.spring.maven.jpa.postgresql.support.AuthenticatedApiIntegrationSupport;

class ResilienceAutoTuneE2ETest extends AuthenticatedApiIntegrationSupport {

    @Test
    void resiliencePoliciesAndResetFlowAreAvailableToAdmins() throws Exception {
        String adminToken = createUserAndLogin("ADMIN");

        mockMvc.perform(get("/api/v1/resilience/circuit-breakers/policies")
                        .header("Authorization", "Bearer " + adminToken))
                .andExpect(status().isOk())
                .andExpect(jsonPath("$.data.items[0].options.threshold").value(5));

        mockMvc.perform(get("/api/v1/resilience/circuit-breakers/search-sync")
                        .header("Authorization", "Bearer " + adminToken))
                .andExpect(status().isOk())
                .andExpect(jsonPath("$.data.name").value("search-sync"));

        mockMvc.perform(post("/api/v1/resilience/circuit-breakers/crawler/reset")
                        .header("Authorization", "Bearer " + adminToken))
                .andExpect(status().isOk())
                .andExpect(jsonPath("$.data.name").value("crawler"));
    }
}
